File: safety_signature_back/src/main/java/com/safety_signature/safety_signature_back/app/common/enumeration/UserTypeCode.java
package com.safety_signature.safety_signature_back.app.common.enumeration;

import com.fasterxml.jackson.annotation.JsonCreator;

/**
 * 회원 유형 코드 ( 마스터 관리자, 기업 관리자, 일반 회원 구분을 위한 코드
 */


public enum UserTypeCode {
    MASTER_ADMINISTRATOR("마스터 관리자 계정"),
    CORPORATE_MANAGER("기업관리자"),
    GENERAL_MEMBER("일반 회원");

    private final String value;

    UserTypeCode(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    @JsonCreator
    public static String from(String sub) {
        for (UserTypeCode code : UserTypeCode.values()) {
            if (code.getValue().equals(sub)) {
                return code.getValue();
            }
        }
        return null;
    }
}

File: safety_signature_app/lib/common/enumeration/user_type_code.dart
enum UserTypeCode {
  MASTER_ADMINISTRATOR("MASTER_ADMINISTRATOR", "마스터 관리자"),
  CORPORATE_MANAGER("CORPORATE_MANAGER", "기업 관리자"),
  GENERAL_MEMBER("GENERAL_MEMBER", "일반 회원"),
  UNDEFINED("UNDEFINED", "");

  const UserTypeCode(this.code, this.displayName);
  final String code;
  final String displayName;
  factory UserTypeCode.getByCode(String code) {
    return UserTypeCode.values.firstWhere((value) => code.contains(value.code),
        orElse: () => UserTypeCode.UNDEFINED);
  }
}
